java/vector/src/main/codegen/templates/DenseUnionVector.java

```diff
@@ -283,7 +285,7 @@ public long getValidityBufferAddress() {
   public ArrowBuf getDataBuffer() { throw new UnsupportedOperationException(); }
 
   public StructVector getStruct(byte typeId) {
-    StructVector structVector = (StructVector) childVectors[typeId];
+    StructVector structVector = typeId < 0 ? null : (StructVector) childVectors[typeId];
```

Comment:
Should negative type IDs throw an explicit error?

Reply:
Ah, negative IDs are used as a sentinel for not yet allocated...

Reply:
correct, its a bit strange as there is no way to tell between null and not yet allocated. One of hte only benefits of a separate validity buffer. I thought this was a good compromise though I dont love it.